code/game/turfs/simulated/floor/chasm.dm

```diff
@@ -294,3 +295,30 @@
 	drop_y = y
 	var/list/target_z = levels_by_trait(SPAWN_RUINS)
 	drop_z = pick(target_z)
+
+/turf/simulated/floor/chasm/space_ruin
+
+/turf/simulated/floor/chasm/space_ruin/drop(atom/movable/AM)
+	//Make sure the item is still there after our sleep
+	if(!AM || QDELETED(AM))
+		return
+	falling_atoms[AM] = TRUE
+
+	var/list/target_z = levels_by_trait(SPAWN_RUINS) // list of all z levels that can get ruins spawned in
+	target_z -= AM.z // excluding the one atom was already in from possible z levels
+	var/turf/T = locate(rand(5, world.maxx - 5), rand(5, world.maxy - 5), pick(target_z))
```

Comment:
```suggestion
	var/turf/T = locate(rand(TRANSITIONEDGE + 1, world.maxx - TRANSITIONEDGE - 1), rand(TRANSITIONEDGE + 1, world.maxy - TRANSITIONEDGE - 1), pick(target_z))
```
`TRANSITIONEDGE` is the number of tiles that will push an atom onto its adjacent z-level, so you'll probably want to avoid that area in case there's any weirdness